chcemvediet/apps/obligees/tests/test_forms.py

```diff
@@ -1,123 +1,175 @@
-from django.core.urlresolvers import reverse
 from django.core.exceptions import ValidationError
 from django.test import TestCase
+from django.utils.translation import ugettext_lazy as _
+
+from poleno.utils.urls import reverse
 
 from chcemvediet.tests import ChcemvedietTestCaseMixin
 
 from ..forms import ObligeeWidget, ObligeeField
 
 
-class ObligeeFieldWithTextInputWidgetTest(ChcemvedietTestCaseMixin, TestCase):
+class ObligeeFieldTest(ChcemvedietTestCaseMixin, TestCase):
     u"""
-    Tests ``ObligeeField`` with ``TextInput`` widget.
+    Tests ``ObligeeField`` field with ``ObligeeWidget`` widget.
     """
 
     class Form(forms.Form):
         obligee = ObligeeField()
 
     class FormWithWidgetAttrs(forms.Form):
         obligee = ObligeeField(
-                widget=forms.TextInput(attrs={
+                widget=ObligeeWidget(input_attrs={
                     u'class': u'custom-class',
                     u'custom-attribute': u'value',
                     }),
                 )
 
 
-    @unittest.skip(u'FIXME')
     def test_new_form(self):
         form = self.Form()
         rendered = self._render(u'{{ form }}', form=form)
+        html = lxml.html.fromstring(rendered)
+
         self.assertInHTML(u'<label for="id_obligee">Obligee:</label>', rendered)
+        root = html.get_element_by_id(u'id_obligee')
+        self.assertEqual(root.tag, u'div')
+        self.assertEqual(root.attrib[u'class'], u'chv-obligee-widget')
+
+        inputs = root.find_class(u'chv-obligee-widget-inputs')[0]
+        self.assertEqual(inputs.tag, u'div')
+        input = inputs.find_class(u'chv-obligee-widget-input')[0]
+        self.assertEqual(input.tag, u'div')
         self.assertInHTML(u"""
-                <input class="autocomplete" data-autocomplete-url="{url}" id="id_obligee" name="obligee" type="text">
+                <input class="form-control pln-autocomplete" data-autocomplete-url="{url}" name="obligee" data-name="obligee" type="text" value="">
                 """.format(url=reverse(u'obligees:autocomplete')), rendered)
 
-    @unittest.skip(u'FIXME')
-    def test_new_form_with_custom_widget_class_attributes(self):
+        details = input.find_class(u'chv-obligee-widget-details chv-obligee-widget-hide')[0]
+        self.assertEqual(details.tag, u'div')
+        street = details.find_class(u'chv-obligee-widget-street')[0]
+        self.assertEqual(street.tag, u'span')
+        self.assertIsNone(street.text)
+        zip = details.find_class(u'chv-obligee-widget-zip')[0]
+        self.assertEqual(zip.tag, u'span')
+        self.assertIsNone(zip.text)
+        city = details.find_class(u'chv-obligee-widget-city')[0]
+        self.assertEqual(city.tag, u'span')
+        self.assertIsNone(city.text)
+        email = details.find_class(u'chv-obligee-widget-email')[0]
+        self.assertEqual(email.tag, u'span')
+        self.assertIsNone(email.text)
+
+    def test_new_form_with_custom_widget_class_and_attributes(self):
```

Comment:
Zrejme tento test treba rozdelit na dva:
1. widget class and attributes, tj. tie co su na toplevel elemente. Zadavaju sa cez `attrs` parameter.
2. **input** class and attributes, tj. tie co su na `input` elemente. Zadavaju sa cez `input_attrs` parameter.

Reply:
:ok: